### PR TITLE
Added instructions for Hawkular APM

### PR DESCRIPTION
**Pull request by jpkrohling**

### CODE REVIEW

**Comment by jpkrohling** [comment]
Note: I believe the `readme.html` should _not_ be on git, but as it was there, I'm sending a change for it as well.

@objectiser : would you please review this? 

@jotak : would you test the instructions, please? Running the "ansible" approach + the "Hawkular APM" part should be enough.

**Comment by jpkrohling** [comment]
@rafabene , would you please check whether the text is in line with the other parts?

File: hawkular-apm.adoc [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+```
+oc env dc/aloha       ZIPKIN_SERVER_URL="http://hawkular-apm-server:8080"
+oc env dc/api-gateway ZIPKIN_SERVER_URL="http://hawkular-apm-server:8080"
+oc env dc/bonjour     ZIPKIN_SERVER_URL="http://hawkular-apm-server:8080"
+oc env dc/hola        ZIPKIN_SERVER_URL="http://hawkular-apm-server:8080"
+oc env dc/ola         ZIPKIN_SERVER_URL="http://hawkular-apm-server:8080"
+```
+
+This allows some services to keep using ZipKin, while other services might prefer/require Hawkular APM.
```
Comment (objectiser) [inline hawkular-apm.adoc]: Not sure we should suggest this, as it won't be good for zipkin or APM to receive a partial view of the services - best that all apps are pointed to one or the other.

File: hawkular-apm.adoc [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+#### Deploy using oc CLI
+
+----
+$ oc new-app jboss/hawkular-apm-server
+$ oc expose service hawkular-apm-server
+----
+
+#### Getting the admin username and password
+
+Once the service is deployed, the admin credentials can be obtained on via Browse → Deployments → hawkular-apm-server → Click on the latest deployment → Logs.
```
Comment (objectiser) [inline hawkular-apm.adoc]: "can be obtained via" - remove 'on'

Comment (jpkrohling) [reply hawkular-apm.adoc]: +1

File: hawkular-apm.adoc (hunk 1 above)
Comment (jpkrohling) [reply hawkular-apm.adoc]: Indeed, I also had my doubts when I wrote this :)

**Comment by jotak** [comment]
will test the ansible instructions with APM

**Comment by jotak** [comment]
Got some "manifest unknown" error when trying to pull the hawkular-apm-server image when running `oc new-app jboss/hawkular-apm-server`:

`2016-09-23T14:37:34+02:00   2016-09-23T14:35:52+02:00   4         hawkular-apm-server-1-7l5cm   Pod       spec.containers{hawkular-apm-server}   Normal    Pulling   {kubelet rhel-cdk}   pulling image "jboss/hawkular-apm-server@sha256:64663f7e4dfba67416dab5568a9176e03330ccc307ab3d5635caba8dcde4343a"
2016-09-23T14:37:38+02:00   2016-09-23T14:35:57+02:00   4         hawkular-apm-server-1-7l5cm   Pod       spec.containers{hawkular-apm-server}   Warning   Failed    {kubelet rhel-cdk}   Failed to pull image "jboss/hawkular-apm-server@sha256:64663f7e4dfba67416dab5568a9176e03330ccc307ab3d5635caba8dcde4343a": manifest unknown: manifest unknown
2016-09-23T14:37:38+02:00   2016-09-23T14:35:57+02:00   4         hawkular-apm-server-1-7l5cm   Pod                 Warning   FailedSync   {kubelet rhel-cdk}   Error syncing pod, skipping: failed to "StartContainer" for "hawkular-apm-server" with ErrImagePull: "manifest unknown: manifest unknown"
`

**Comment by objectiser** [comment]
Looks like a known issue: https://blog.openshift.com/openshift-and-manifest-unknown/

Other approach is to clone the Hawkular dockerfiles: https://github.com/jboss-dockerfiles/hawkular and deploy the same as the other MSA services?

**Comment by rafabene** [comment]
I've tried this PR but I faced the following issue when deploy hawkular: Failed to pull image "jboss/hawkular-apm-server@sha256:64663f7e4dfba67416dab5568a9176e03330ccc307ab3d5635caba8dcde4343a": manifest unknown: manifest unknown

It seems that this hawkular docker image uses the new format that it's not yet supported by OSE 3.2. Can you verify that, please? How did you do to run hawkular in CDK?

Thanks

**Comment by objectiser** [comment]
@rafabene Yes we had this problem too, as described in my previous comment. So until openshift supports this approach, we will have to update the PR to describe cloning the hawkular dockerfile repo and use the "oc new-build", "oc new-app" approach.

Juca will be back from PTO soon, so will be able to fix the PR. But if you could give the alternate approach a try that would be great.

**Comment by objectiser** [comment]
This PR can be closed.

**Pull request closed without merging**